src/interface.rs

```diff
@@ -1078,6 +1136,60 @@ mod tests {
         });
     }
 
+    #[test]
+    fn test_autoip() {
+        testing_logger::setup();
```

Comment:
Why this test logger if it's never really used in the test? But I would suggest also maybe a unit test that tests the `has_undhandled_field()`/`check_extra_flags()`.